### PR TITLE
Pixel shift

### PR DESCRIPTION
**Pull request by EmileSonneveld**
https://github.com/Open-EO/openeo-geopyspark-driver/issues/648

The stac catalog in this PR can be loaded with this URL: https://raw.githubusercontent.com/Open-EO/openeo-geopyspark-driver/refs/heads/pixel_shift/tests/data/stac/issue648-pixel-shift/collection.json 

This MR failed to build a few times, but started working again by itself. It failed and worked on jenkinsslave7.

### CODE REVIEW

File: openeogeotrellis/load_stac.py [hunk 1]
```diff
@@ -496,6 +496,10 @@ def intersects_temporally(interval) -> bool:
     metadata_properties = {}
     correlation_id = env.get(EVAL_ENV_KEY.CORRELATION_ID, "")
 
+    if load_params.global_extent is None or len(load_params.global_extent) == 0:
+        # Layer catalog global extent is not known at the moment load_params is created.
+        load_params = load_params.copy()
+        load_params.global_extent = stac_bbox.as_dict()
```
Comment (EmileSonneveld) [inline openeogeotrellis/load_stac.py:502]: @bossie A better way would be to do a dry-run to create a new load_params object. Or a virtual layer in the layer catalog. 
Would this be an ok solution for now?

Comment (EmileSonneveld) [reply openeogeotrellis/load_stac.py:502]: global_extent is needed when calling `datacube_parameters.create()` in the next line. It is used for pixel perfect alignment with source data

**Comment by EmileSonneveld** [comment]
The stac catalog in this PR can be loaded with this URL: https://raw.githubusercontent.com/Open-EO/openeo-geopyspark-driver/refs/heads/pixel_shift/tests/data/stac/issue648-pixel-shift/collection.json

File: openeogeotrellis/load_stac.py (hunk 1 above)
Comment (jdries) [reply openeogeotrellis/load_stac.py:502]: but how can there be alignment if we just use the local stac bbox?
My suggestion was indeed to properly track the stac collection bounding boxes during dry run, allowing to do a real global extent construction, just like we do for load_collection.

Comment (EmileSonneveld) [reply openeogeotrellis/load_stac.py]: global_extent for load_stac is not read from cube:dimensions

**Pull request merged**